src/snowcli/cli/nativeapp/version/version_processor.py

```diff
@@ -0,0 +1,311 @@
+        repo = Repo(project_root)
+
+        # Check if the repo has any changes, including untracked files
+        if repo.is_dirty(untracked_files=True):
+            print("Changes detected in your git repository!")
+            # show differences between current files and last commit
+            print(repo.git.diff(repo.head.commit.tree))
+
+            ask_user_confirmation(
+                user_confirmation_policy=user_confirmation_policy,
+                prompts=Prompts(
+                    f"You have local changes in this repository that are not part of a previous commit. Do you still want to continue? Consider using 'git status' and/or 'git diff' before proceeding.",
+                    f"Did not create a new version {version} of application package {package_name}.",
+                    "Cannot create a new version without user input. Consider using --force if you would like to attempt forcing the creation of version.",
+                ),
+            )
+
+    except InvalidGitRepositoryError:
+        pass  # not a git repository, which is acceptable
+
+
+def warn_user_about_existing_rd(
+    existing_rds: List[dict],
+    version: str,
+    package_name: str,
+    user_confirmation_policy: UserConfirmationPolicy,
+) -> None:
+    """
+    Warns the user if a version is already referenced in a release directive(s), and asks for confirmation to add a patch.
+    """
+    if existing_rds:
+        rd_names = ", ".join(row["name"] for row in existing_rds)
+        print(
+            dedent(
+                f"""\
+                Version {version} already exists for application package {package_name} and in release directive(s): {rd_names}.
+            """
+            )
+        )
+        ask_user_confirmation(
+            user_confirmation_policy=user_confirmation_policy,
+            prompts=Prompts(
+                f"Are you sure you want to create a new patch for version {version} of application package {package_name}? Once added, this operation cannot be undone.",
+                f"Did not create a new patch for version {version} of application package {package_name}.",
+                "Cannot create a new patch without user input. Consider using --force if you would like to attempt forcing the creation of patch.",
+            ),
+        )
+
+
+class NativeAppVersionCreateProcessor(NativeAppRunProcessor):
+    def __init__(self, project_definition: Dict, project_root: Path):
+        super().__init__(project_definition, project_root)
+
+    def get_existing_version_info(self, version: str) -> Optional[dict]:
+        """
+        Check for an existing version by the same name for an application package.
+        It executes a 'show versions like ... in application package' query and returns the result as single row, if one exists.
+        """
+        with self.use_role(self.package_role):
+            show_obj_query = f"show versions like '{unquote_identifier(version)} in application package {self.package_name}"
```

Comment:
Missing end quote. Also I cannot remember off-the-top of my head: do versions follow the same quoting rules as everything else?

Reply:
Addressed. 
Re: quoting rules - I am not sure myself. I'll need to check this.